lib/engines/hcktest/tests.rb

```diff
@@ -132,11 +132,37 @@ def wait_queued_test(id)
       end
     end
 
-    def test_parameters(test_name)
+    def read_test_config(test_name, config)
       @project.engine.config.tests_config
               .select { _1.tests.include?(test_name) }
-              .flat_map(&:parameters)
-              .to_h { |parameter| [parameter.name, parameter.value] }
+              .flat_map(&config)
+    end
+
+    def test_parameters(test_name)
+      read_test_config(test_name, :parameters)
+        .to_h { |parameter| [parameter.name, parameter.value] }
+    end
+
+    def run_guest_commands(test, commands)
```

Comment:
Here we only give one command for each call so the parameter should be singular.